shared/actions/fs/index.js

```diff
@@ -575,21 +575,25 @@ const editSuccess = (res, action, state: TypedState) => {
 const editFailed = (res, {payload: {editID}}) => Saga.put(FsGen.createEditFailed({editID}))
 
 function* openPathItem(action: FsGen.OpenPathItemPayload): Saga.SagaGenerator<any, any> {
-  const {path, routePath} = action.payload
+  const {path, routePath, openDirectly} = action.payload
   const state: TypedState = yield Saga.select()
   const pathItem = state.fs.pathItems.get(path)
   if (!pathItem || pathItem.type === 'folder') {
-    yield Saga.put(
-      putActionIfOnPath(
-        routePath,
-        navigateAppend([
-          {
-            props: {path},
-            selected: 'folder',
-          },
-        ])
+    const navigate = navigateAppend([{
```

Comment:
If you decide to make a separate action for this, we can group the `switchTo` with `navigateAppend` here, rather than needing to dispatch multiple actions in the remote container.